internal/streamingnode/server/wal/adaptor/message_handler.go

```diff
@@ -125,7 +127,9 @@ func (m *MsgPackAdaptorHandler) addMsgPackIntoPending(msgs ...message.ImmutableM
 	}
 }
 
-// Close close the handler.
+// Close closes the handler.
 func (m *MsgPackAdaptorHandler) Close() {
-	close(m.channel)
+	m.closeOnce.Do(func() {
```

Comment:
we should not add a rerundant once here.
if the channel is closed twice, it's just a bug, we should fix it.